metabase/metabase
# Add default delete action to actions dropdown in object detail modal

Closes #32323

### Description

- Tests will be added in #32350

### How to verify

> Clicking delete should open a confirmation modal where the user has to confirm that they wish to delete the record and that there’s no undo.

> If a successful delete is run from an object detail modal, the detail modal should close and the user should see a toast. The row that was deleted should no longer be present in the table.

### Design notes
- The object detail modal has up/down arrows to navigate between neighboring rows. We don't need to close the object detail modal after deleting the object - we can navigate to next or previous object, or close the modal if there are no more objects. I believe [this was the original intention](https://www.notion.so/metabase/Allow-updates-and-deletes-on-model-object-detail-views-f05b298c3aba4a6ca883b2738d86824d?pvs=4#1ff1b9c44cd544d5b60881f80bcf272f) for the descoped part with objects displayed in dashboards. This PR implements closing both modals when deleting objects. If you want to try out how it would have worked with auto-navigation to sibling row please `git cherry-pick 8b24685eee5cd968f53cc14b7194a66588fb5845` into this branch
- Deleting an object that is referenced by other objects will often result in an error ("Referential integrity constraint violation") - see the end of demo - we might want to check if it's possible to hide/disable deleting in that case beforehand. Or consider cascade delete option?

### Demo

This demo uses 2 simple models that do nothing but reference sample db tables: accounts & reviews respectively.

https://github.com/metabase/metabase/assets/6830683/6b954457-1aa1-41c7-b154-26ffd05baaca
